db/sql/queries/web_vulnerability.sql

```diff
@@ -21,3 +21,11 @@ WHERE vulnerability_id = $1;
 SELECT *
 FROM web_vulnerabilities
 WHERE scan_id = $1;
+
+-- name: ListWebVulnerabilitiesByServiceID :many
+SELECT I.uri, I.method, I.param, I.evidence, I.other_info, I.attack, W.vulnerability_id, W.scan_id, W.host_id, W.service_id, W.solution_advice, W.reference, W.created_at, W.updated_at
+FROM (SELECT * FROM web_vulnerabilities WHERE service_id=$1) W
+INNER JOIN (SELECT id FROM vulnerabilities WHERE vuln_type ='WEB_APPLICATION') V
+ON W.vulnerability_id=V.id
+INNER JOIN web_vulnerability_instances I ON W.vulnerability_id=I.source_vulnerability_id;
```

Comment:
This SQL query uses an `INNER JOIN` on `web_vulnerability_instances`.

If a `web_vulnerability` exists but has **zero** instances associated to it, the `INNER JOIN` will fail to find a match, and that vulnerability will be excluded. We would silently miss data.

Now business logic wise this shouldn't be the case, since instances typically exist for a web vulnerability... but I may be mistaken. It might be safer to go with a `LEFT JOIN` here.


```suggestion
LEFT JOIN web_vulnerability_instances I ON W.vulnerability_id=I.source_vulnerability_id;
```

which would fill up columns from the `instances` table as `NULL`

Reply:
Yes I agree I will change it